frontend/src/pages/CardDetail.js

```diff
@@ -75,7 +75,7 @@ class CardDetail extends Component {
                       Save Card Preferences
                     </Button>
                   </div>
-                ) : (
+                ) : cardDetail.user == null ? (
```

Comment:
I'm not sure about this. Realistically if a card exists then it should have a owner, so I don't think this should ever actually be null, it just might be with your db since we are working with fake data.

I think if a card is for sale it still has an owner and that owner would be the seller right? So if you do this and user 2 is selling the card, and user 1 views it, then the buy button will not be shown.

You might have to get listing information from the backend to check if the card is for sale.